diff --git a/dist/index.js b/dist/index.js
index 6806be1e729e0d348b0cc8890e87236e71331980..7e2fab90208e31b4bd37dcc997e51b6879c69bb5 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -49,7 +49,7 @@ function createSlot(ownerName) {
     let slottableElement = null;
     let hasSlottable = false;
     const newChildren = [];
-    if (isLazyComponent(children) && typeof use === "function") {
+    while (isLazyComponent(children) && typeof use === "function") {
       children = use(children._payload);
     }
     React.Children.forEach(children, (maybeSlottable) => {
@@ -57,7 +57,7 @@ function createSlot(ownerName) {
         hasSlottable = true;
         const slottable = maybeSlottable;
         let child = "child" in slottable.props ? slottable.props.child : slottable.props.children;
-        if (isLazyComponent(child) && typeof use === "function") {
+        while (isLazyComponent(child) && typeof use === "function") {
           child = use(child._payload);
         }
         slottableElement = getSlottableElementFromSlottable(slottable, child);
diff --git a/dist/index.mjs b/dist/index.mjs
index 217af5391d66a92241b546d3435647cfc2669c2f..3cb6384f8711f1d592b090376b83776e2a1f83cc 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -11,7 +11,7 @@ function createSlot(ownerName) {
     let slottableElement = null;
     let hasSlottable = false;
     const newChildren = [];
-    if (isLazyComponent(children) && typeof use === "function") {
+    while (isLazyComponent(children) && typeof use === "function") {
       children = use(children._payload);
     }
     React.Children.forEach(children, (maybeSlottable) => {
@@ -19,7 +19,7 @@ function createSlot(ownerName) {
         hasSlottable = true;
         const slottable = maybeSlottable;
         let child = "child" in slottable.props ? slottable.props.child : slottable.props.children;
-        if (isLazyComponent(child) && typeof use === "function") {
+        while (isLazyComponent(child) && typeof use === "function") {
           child = use(child._payload);
         }
         slottableElement = getSlottableElementFromSlottable(slottable, child);
